<launch>

  <node pkg="base_robot" type="robot_differential_node.py" name="nodo_robot_diferencial" output="screen">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baudrate" value="57600"/>
    <param name="wheel_radius" value="0.065"/>
    <param name="wheel_base" value="0.42"/>
    <param name="encoder_ticks" value="341"/>
  </node>

<node pkg="base_robot" type="dual_camera.py" name="nodo_Camaras" output="screen">
    <param name="front_camera_topic" value="/camera/front/image_raw" />
    <param name="rear_camera_topic" value="/camera/rear/image_raw" />
  </node>

<node pkg="base_robot" type="busqueda2.py" name="nodo_busqueda" output="screen">
  </node>

<node pkg="base_robot" type="descarga.py" name="nodo_descarga" output="screen">
  </node>

  <node name="nodo_sensor_lidar"  pkg="ydlidar_ros_driver"  type="ydlidar_ros_driver_node" output="screen" respawn="false" >
    <!-- string property -->
    <param name="port"         type="string" value="/dev/ydlidar"/>  
    <param name="frame_id"     type="string" value="laser_frame"/>
    <param name="ignore_array"     type="string" value=""/>

    <!-- int property -->
    <param name="baudrate"         type="int" value="128000"/>  
    <!-- 0:TYPE_TOF, 1:TYPE_TRIANGLE, 2:TYPE_TOF_NET -->
    <param name="lidar_type"       type="int" value="1"/>  
    <!-- 0:YDLIDAR_TYPE_SERIAL, 1:YDLIDAR_TYPE_TCP -->
    <param name="device_type"         type="int" value="0"/>  
    <param name="sample_rate"         type="int" value="5"/>  
    <param name="abnormal_check_count"         type="int" value="4"/>  

    <!-- bool property -->
    <param name="resolution_fixed"    type="bool"   value="true"/>
    <param name="auto_reconnect"    type="bool"   value="true"/>
    <param name="reversion"    type="bool"   value="false"/>
    <param name="inverted"    type="bool"   value="true"/>
    <param name="isSingleChannel"    type="bool"   value="false"/>
    <param name="intensity"    type="bool"   value="false"/>
    <param name="support_motor_dtr"    type="bool"   value="true"/>
    <param name="invalid_range_is_inf"    type="bool"   value="false"/>
    <param name="point_cloud_preservative"    type="bool"   value="false"/>

    <!-- float property -->
    <param name="angle_min"    type="double" value="-180" />
    <param name="angle_max"    type="double" value="180" />
    <param name="range_min"    type="double" value="0.1" />
    <param name="range_max"    type="double" value="12.0" />
    <!-- frequency is invalid, External PWM control speed -->
    <param name="frequency"    type="double" value="10.0"/>
  </node>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser"
    args="0.0 0.0 0.2 0.0 0.0 0.0 /base_footprint /laser_frame 20" />
</launch>
